### ISSUE TITLE
[kernel-heap] can't find crate for `alloc`

### ISSUE THREAD

**Issue by kyrias** [phil-opp/blog_os#312]
Trying to compile the code from the kernel-heap post, both what I've written myself and what's in the repo under the `kernel_heap` tag, gives the following error on a current nightly:

```
error[E0463]: can't find crate for `alloc`
  --> src/lib.rs:26:1
   |
26 | extern crate alloc;
   | ^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error
```

**Comment by phil-opp** [comment]
Thanks for the report! 

The problem is that `xargo` only cross compiles the `core` library by default, but we also need the `alloc` and `collection` libraries. To fix this, you need to create a file named `Xargo.toml` at the project root with the following content:

```toml
[target.x86_64-blog_os.dependencies]
collections = {}
```
This instructs `xargo` to also cross compile `collections` and `alloc`.

---

This solution was mentioned in the old “Returning from Exceptions” post, but I forgot to add it to the “Kernel Heap” post after we landed #300. I try to fix it in the next few days.

**Comment by kyrias** [comment]
Ah, I see, thanks!
